src/common/KokkosKernels_Sorting.hpp

```diff
@@ -383,191 +1023,21 @@ TeamBitonicSort2(ValueType* values, PermType* perm, Ordinal n, const TeamMember
-          [&](const Ordinal i)
-          {
-            Ordinal globalThread = i + t.league_rank() * work;
-            Ordinal subBox = globalThread >> (logSubBoxSize - 1);
-            Ordinal subBoxStart = subBox << logSubBoxSize;
-            Ordinal subBoxOffset = globalThread & ((Ordinal(1) << (logSubBoxSize - 1)) - 1); //i % (subBoxSize / 2)
-            Ordinal elem1 = subBoxStart + subBoxOffset;
-            //later phases (pink box): within a block, compare with fixed distance (boxSize / 2) apart
-            Ordinal elem2 = elem1 + subBoxSize / 2;
-            if(elem2 < Ordinal(v.extent(0)))
-            {
-              if(comp(v(elem2), v(elem1)))
-              {
-                Value temp = v(elem1);
-                v(elem1) = v(elem2);
-                v(elem2) = temp;
-              }
-            }
-          });
-      }
-    }
-  };
-  View v;
-  Comparator comp;
-  Ordinal boxSize;
-  Ordinal teamsPerBox;
-};
-
-//Version to be called from host on a single array
-//Generally ~2x slower than Kokkos::sort() for large arrays (> 50 M elements),
-//but faster for smaller arrays.
-//
-//This is more general than BinSort: bitonic supports any trivially copyable type
-//and an arbitrary device-compatible comparison operator (provided through operator() of Comparator)
-//If comparator is void, use operator< (which should only be used for primitives)
-template<typename View, typename ExecSpace, typename Ordinal, typename Comparator = DefaultComparator<typename View::value_type>>
-void bitonicSort(View v, const Comparator& comp = Comparator())
-{
-  typedef Kokkos::TeamPolicy<ExecSpace> team_policy;
-  typedef typename team_policy::member_type team_member;
-  Ordinal n = v.extent(0);
-  //If n is small, just sort on a single team
-  if(n <= Ordinal(1) << 12)
-  {
-    Kokkos::parallel_for(team_policy(1, Kokkos::AUTO()),
-        BitonicSingleTeamFunctor<View, Ordinal, team_member, Comparator>(v, comp));
-  }
-  else
-  {
-    Ordinal npot = 1;
-    while(npot < n)
-      npot <<= 1;
-    //Partition the data equally among fixed number of teams
-    Ordinal chunkSize = 512;
-    Ordinal numTeams = npot / chunkSize;
-    //First, sort within teams
-    Kokkos::parallel_for(team_policy(numTeams, Kokkos::AUTO()),
-        BitonicChunkFunctor<View, Ordinal, team_member, Comparator>(v, comp, chunkSize));
-    for(int teamsPerBox = 2; teamsPerBox <= npot / chunkSize; teamsPerBox *= 2)
-    {
-      Ordinal boxSize = teamsPerBox * chunkSize;
-      Kokkos::parallel_for(team_policy(numTeams, Kokkos::AUTO()),
-          BitonicPhase1Functor<View, Ordinal, team_member, Comparator>(v, comp, boxSize, teamsPerBox));
-      for(int boxDiv = 1; teamsPerBox >> boxDiv; boxDiv++)
-      {
-        Kokkos::parallel_for(team_policy(numTeams, Kokkos::AUTO()),
-            BitonicPhase2Functor<View, Ordinal, team_member, Comparator>(v, comp, boxSize >> boxDiv, teamsPerBox >> boxDiv));
-      }
-    }
-  }
+  using KokkosKernels::sort_crs_graph;
```

Comment:
Are these imported in the `impl` namespace to avoid downstream failure? If that is the case I would argue that we want to also mark these as deprecated so we can eventually remove them and only keep the public interface outside of the `impl` namespace

Reply:
Good idea, done.